src/lib.rs

```diff
@@ -133,15 +134,31 @@ impl CompiledProgram {
     /// - Witness values have a different type than declared in the Simfony program.
     /// - There are missing witness values.
     pub fn satisfy(&self, witness_values: WitnessValues) -> Result<SatisfiedProgram, String> {
+        self.satisfy_with_env(witness_values, None)
+    }
+
+    /// Satisfy the Simfony program with the given `witness_values`.
+    /// If `env` is `None`, the program is not pruned, otherwise it is pruned with the given environment.
+    ///
+    /// ## Errors
+    ///
+    /// - Witness values have a different type than declared in the Simfony program.
+    /// - There are missing witness values.
```

Comment:
30f125f18942a096edef794df7c0c06ea42345bd: nit: pruning fails if the program doesn't run successfully for the given environment and witness data.